Rubberduck.Parsing/Grammar/VBAParser.g4

```diff
@@ -877,7 +878,8 @@ commentOrAnnotation :
 ;
 remComment : REM whiteSpace? commentBody;
 comment : SINGLEQUOTE commentBody;
-commentBody : (LINE_CONTINUATION | ~NEWLINE)*;
+// commentBody must terminate with a NEWLINE, see VBA Language spec, section 3.3.1 and 3789
+commentBody : (LINE_CONTINUATION | ~NEWLINE)* (NEWLINE | EOF);
 annotationList : SINGLEQUOTE (AT annotation whiteSpace?)+ (whiteSpace? COLON commentBody)?;
```

Comment:
This is inconsistent now. If there is no trailing comment, the `NEWLINE` will not get consumed.

Reply:
What do you think of the following solution?

``` 
annotationList : SINGLEQUOTE (AT annotation whiteSpace?)+ ((COLON commentBody?) | (NEWLINE | EOF));
``` 

Of note: I removed the redundant `whiteSpace?` before the `COLON` and the logical LineEnd is definitely consumed.  
It might be useful to define `(NEWLINE | EOF)` as `logicalLineEnd` to avoid repetition there...